test/grains_cases_test.rb

```diff
@@ -0,0 +1,87 @@
+require_relative 'test_helper'
+
+class GrainsTest < Minitest::Test
+  def test_test_name
+    test_case = GrainsCase.new(description: 'description')
+
+    assert_equal 'test_description', test_case.test_name
+  end
+
+  def test_test_name_when_description_has_spaces
+    test_case = GrainsCase.new(description: 'description with spaces')
+
+    assert_equal 'test_description_with_spaces', test_case.test_name
+  end
+
+  def test_workload_is_not_implemented
+    assert_raises(NotImplementedError) { GrainsCase.new.workload }
+  end
+
+  def test_skipped_with_zero_index
+    assert_equal '# skip', GrainsCase.new(index: 0).skipped
+  end
+
+  def test_skipped_with_nil_index
```

Comment:
Getting closer :smile:

In the above example, how does `i` relate to `cases.size`?